src/mango_cursor.erl

```diff
@@ -30,29 +31,20 @@
 
 create(Db, Selector0, Opts) ->
     Selector = mango_selector:normalize(Selector0),
-
-    ExistingIndexes = mango_idx:list(Db),
-    if ExistingIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, no_indexes_defined})
-    end,
-
-    FilteredIndexes = maybe_filter_indexes(ExistingIndexes, Opts),
-    if FilteredIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, no_index_matching_name})
-    end,
-
-    SortIndexes = mango_idx:for_sort(FilteredIndexes, Opts),
-    if SortIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, missing_sort_index})
-    end,
-
-    UsableFilter = fun(I) -> mango_idx:is_usable(I, Selector) end,
-    UsableIndexes = lists:filter(UsableFilter, SortIndexes),
-    if UsableIndexes /= [] -> ok; true ->
-        ?MANGO_ERROR({no_usable_index, selector_unsupported})
-    end,
-
-    create_cursor(Db, UsableIndexes, Selector, Opts).
+    UsableIndexes = mango_idx:get_usable_indexes(Db, Selector0, Opts),
+
+    {use_index, IndexSpecified} = proplists:lookup(use_index, Opts),
+    case {length(UsableIndexes), length(IndexSpecified)} of
+        {0, 1} ->
+            ?MANGO_ERROR({no_usable_index, selector_unsupported});
+        {0, 0} ->
+            AllDocs = mango_idx:special(Db),
+            io:format("Index ~p~n", [AllDocs]),
+            create_cursor(Db, AllDocs, Selector, Opts);
```

Comment:
@robertkowalski:

So this is an interesting case because we need to manually construct the range. In your previous commit, when you do 
```erlang
{[{<<"$and">>, [Selector, {[{<<"_id">>, {[{<<"$gt">>, null}]}}]}]}]}
```
that's essentially doing the same manual range construction, but we are modifying the selector. 

In order to do it without modifying the selector you can:

1) Add some case statements in here and manually set the range in the end.
https://github.com/apache/couchdb-mango/blob/master/src/mango_cursor_view.erl#L31-L49
(I don't like this because it breaks the abstraction layer).

2) Create a new module mango_cursor_special, with a new create function that specifically creates "all_docs" cursor with the range set. However, you'd have to modify these lines to incorporate it:

https://github.com/apache/couchdb-mango/blob/master/src/mango_idx.erl#L247-L248
https://github.com/apache/couchdb-mango/blob/master/src/mango_cursor.erl#L127-L130

At execution time, it will now call mango_cursor_special:execute, but I inside there, you can just defer the call to mango_cursor_view:execute.
I like this because it's cleaner and more modularized. 

Let me know if that suggestion helps.